docs/docsite/rst/dev_guide/migrating_roles.rst

```diff
@@ -0,0 +1,407 @@
+collection without duplicating them in each role's :file:`library/`` directory.
+
+You also need to migrate roles to collections if you want them as part of certified Ansible content.
+
+See :ref:`developing_collections` for details on collections.
+
+
+.. contents::
+   :local:
+   :depth: 1
+
+Comparing standalone roles to collection roles
+===============================================
+
+:ref:`Standalone roles <playbooks_reuse_roles>` have the following directory structure:
+
+.. code-block:: bash
+  :emphasize-lines: 5,7,8
+
+    role/
+    ├── defaults
+    ├── files
+    ├── handlers
+    ├── library
+    ├── meta
+    ├── module_utils
+    ├── [*_plugins]
+    ├── tasks
+    ├── templates
+    ├── tests
+    └── vars
+
+
+The highlighted directories above will change when you migrate to a collection-based role. The collection directory structure includes a :file:`roles/` directory:
+
+.. code-block:: bash
+
+    mynamespace/
+    └── mycollection/
+      ├── docs/
+      ├── galaxy.yml
+      ├── plugins/
+      │   ├── modules/
+      │   │   └── module1.py
+      │   ├── inventory/
+      │   └── .../
+      ├── README.md
+      ├── roles/
+      │   ├── role1/
+      │   ├── role2/
+      │   └── .../
+      ├── playbooks/
+      │   ├── files/
+      │   ├── vars/
+      │   ├── templates/
+      │   └── tasks/
+      └── tests/
+
+You will need to use the Fully Qualified Collection Name (FQCN)` when you migrate your role into a collection. The FQCN is the combination of the collection ``namespace``, collection ``name``, and the content item you are referring to.
+
+So for example, in the above collection, the FQCN to access  ``role1`` would be:
+
+.. code-block:: Python
+
+	mynamespace.mycollection.role1
+
+
+A collection can contain one or more roles in the :file:`roles/` directory and these are almost identical to standalone roles, except you need to move plugins out of the individual roles, and use the :abbr:`FQCN (Fully Qualified Collection Name)` in some places, as detailed in the next section.
+
+.. note::
+
+	 In standalone roles, some of the plugin directories referenced their plugin types in the plural sense; this is not the case in collections.
+
+.. _simple_roles_in_collections:
+
+Migrating a role to a collection
+=================================
+
+To migrate from a standalone role (with no plugins) to a collection role:
+
+1. Create a local :file:`ansible_collections` directory and ``cd`` to this new directory.
+
+2. Create a collection. You need a `Galaxy namespace <https://galaxy.ansible.com/docs/contributing/namespaces.html>`_ to import this collection to Galaxy.
```

Comment:
I know we talked about this previously—but as this guide seems to be for any role (and not just roles that are already on Galaxy)—should this be a top level requirement, or can we move it to an aside? If an aside, I'd suggest:

> ```
> If you wish to share your collection on Ansible Galaxy, you will need a `Galaxy namespace <https://galaxy.ansible.com/docs/contributing/namespaces.html>` to import the collection. Make sure you use the same namespace when creating the collection.
> ```

Something along those lines.